examples/tests/sdk_tests/queue/queue-ref.test.w

```diff
@@ -0,0 +1,54 @@
+bring cloud;
+bring aws;
+bring util;
+
+// a really cool way to test a "ref" resource: basically, we define a cloud.Queue
+// and then we define a `QueueRef` that references it's ARN. nice, ha?
+
+let c = new cloud.Counter();
+let q = new cloud.Queue();
+
+q.setConsumer(inflight () => {
+  c.inc();
+});
+
+// this will only work if we are testing on tf-aws
+if let arn = aws.Queue.from(q)?.queueArn {
+  let qref = new aws.QueueRef(arn);
+
+  new cloud.Function(inflight () => {
+    qref.push("hello");
+    qref.pop();
+  });
+
+  test "can push to an external queue (QueueRef)" {
+    qref.push("m1");
+    qref.push("m2");
+    qref.push("m3");
+    util.waitUntil(() => {
+      return c.peek() == 3;
+    });
```

Comment:
Random design thoughts - should `util.waitUntil` throw by default if the condition is never met? (Right now there's a default timeout of around a minute, and if it fails, then it just returns `false`). And perhaps the throwing could be disabled with `throw: false`?

Reply:
Mmm? It's supposed to throw after the timeout... Otherwise what's the point?

Reply:
OK, I am looking at the code and you are right. It's not throwing. Of course it needs to throw!

Reply:
https://github.com/winglang/wing/issues/6006